tests/test_date.py

```diff
@@ -2,90 +2,158 @@
+    def when_date_range_generated(self):
+        self.range = list(date.date_range(self.period_start, self.period_end))
 
-    def test_should_reject_easily_mistaken_dateutil_arguments(self):
-        begin = datetime(2014, 6, 15)
-        end = datetime(2014, 6, 25)
+    def given_period_size(self, **params):
+        self.period_size = params
 
+    def when_month_range_generated(self):
+        self.generated_months = list(date.date_range(self.period_start, self.period_end, **self.period_size))
+
+    def then_month_expected(self, expected):
+        self.assertEqual(expected,
+                         [(d.year, d.month) for d in self.generated_months])
+
+    def then_range_length_is(self, expected_length):
+        self.assertEqual(expected_length, len(self.range))
+
+    def then_all_dates_from_beginning_to_end_are_present_in_range(self):
+        date_under_test = self.period_start
+        while date_under_test < self.period_end:
+            self.assertIn(date_under_test, self.range)
+            date_under_test += timedelta(days=1)
+
+    def then_range_is_in_ascending_order(self):
+        for i in xrange(len(self.range) - 1):
+            self.assertLess(self.range[i], self.range[i + 1])
+
+    def then_should_reject_easily_mistaken_dateutil_arguments(self):
         with self.assertRaisesRegexp(ValueError, "Invalid argument"):
-            date.date_range(begin, end, month=1).next()
+            date.date_range(self.period_start, self.period_end, month=1).next()
 
         with self.assertRaisesRegexp(ValueError, "Invalid argument"):
-            date.date_range(begin, end, day=1).next()
+            date.date_range(self.period_start, self.period_end, day=1).next()
 
 
 class GetIntersectingPeriodsTest(BaseTestCase):
+    def setUp(self):
+        super(GetIntersectingPeriodsTest, self).setUp()
+        self.intersected_dates = NotImplemented
+        self.expected_results = NotImplemented
 
-    def test_date_arguments_and_date_range_with_default_post_days(self):
-        low = datetime(2014, 6, 15)
-        high = datetime(2014, 6, 16)
-        dates = list(date.get_intersecting_periods(low, high))
-        self.assertEquals(1, len(dates))
-        self.assertEquals(low, dates[0])
+    @parameterized.expand([
+        param(low=datetime(2014, 6, 15), high=datetime(2014, 6, 16),
+              dates=list(date.get_intersecting_periods(datetime(2014, 6, 15), datetime(2014, 6, 16))))
+    ])
+    def test_date_arguments_and_date_range_with_default_post_days(self, low, high, dates):
+        self.given_period_low(low)
+        self.given_period_high(high)
+        self.when_date_intersecting_generated(dates)
+        self.then_date_period_and_intersected_dates_should_equal()
+        self.then_date_arguments_and_date_range_length_should_equal()
 
     @parameterized.expand([
         param(low=datetime(2014, 4, 15),
               high=datetime(2014, 6, 25),
+              period='month',
```

Comment:
You see how test is called `test_should_one_date_for_each_month` - we use `month` as a part of test behavior and don't need to parametrize it.
Another thing as if you look at this test and `test_periods` you'll see that their behaviour is exactly the same. So you can merge the data (and then keeping period in param would be appropriate and expected)